base/deprecated.jl

```diff
@@ -164,6 +164,10 @@ function promote_eltype_op end
 
 # @deprecate one(i::CartesianIndex) oneunit(i)
 # @deprecate one(::Type{I}) where I<:CartesianIndex oneunit(I)
+
+@deprecate reindex(V, idxs, subidxs) reindex(idxs, subidxs)
```

Comment:
I think these should be commented out like the lines above it, since we can't introduce new deprecations until 2.0, if I'm not mistaken.

Reply:
These are internal functions, so they don't need deprecations at all unless we're aware that people are using them and we want to be nice.

Reply:
I'm not aware of any users, but this seemed harmless and better than throwing a `MethodError`.

Reply:
BTW `_indices_sub` is not deprecatable, so this doesn't try.

Reply:
Friendly reminder to use a trailing `false` in cases like this to prevent unexported things from being exported. :)

#31201